File: src/1582.special-positions-in-a-binary-matrix.1704/solution.ts
function numSpecial (mat: number[][]): number {
    const M:number = mat.length;
    if (M === 0) {
        return 0;
    }
    const N:number = mat[0].length;
    const vertical:number[] = new Array(M).fill(0);
    const horizontal:number[] = new Array(N).fill(0);
    for (let i = 0; i < M; i++) {
        for (let j = 0; j < N; j++) {
            if (mat[i][j] === 1) {
                vertical[i]++;
                horizontal[j]++;
            }
        }
    }
    let result = 0;
    for (let i = 0; i < M; i++) {
        if (vertical[i] !== 1) {
            continue;
        }
        for (let j = 0; j < N; j++) {
            if (mat[i][j] === 1 && horizontal[j] === 1) {
                result++;
            }
        }
    }
    return result;
}

File: src/1582.special-positions-in-a-binary-matrix.1704/solution.java
class Solution {
    public int numSpecial(int[][] mat) {
        int M = mat.length;
        int N = mat[0].length;
        int[] rows = new int[M];
        int[] columns = new int[N];
        for(int i=0;i<M;i++){
            for(int j=0;j<N;j++){
                rows[i] += mat[i][j];
                columns[j] += mat[i][j];
            }
        }
        int result = 0;
        for(int i=0;i<M;i++){
            if(rows[i] != 1){
                continue;
            }
            for(int j=0;j<N;j++){
                if(mat[i][j] ==1 && columns[j] == 1){
                    result++;
                }
            }
        }
        return result;
    }
}